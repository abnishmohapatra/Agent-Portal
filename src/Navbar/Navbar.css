body {
    margin: 0;
    font-family: Arial, sans-serif;
}

.navbar {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background: #DFDFF6;
    padding: 10px 20px;
    color: white;
}

.navbar-left,
.navbar-right {
    display: flex;
    align-items: center;
}

.menu-icon {
    font-size: 24px;
    cursor: pointer;
    margin-right: 20px;
}

.logo {
    height: 50px;
    margin-right: 9%;
    margin-left: 15%;
}

.greeting {
    font-size: 2rem;
    color: black;
}



.profile-image {
    height: 40px;
    width: 40px;
    border-radius: 50%;
    margin-right: 20px;
}

h1 {
    font-size: 1rem;
    margin-right: 20px;
    color: black;
}

.action-button {
    padding: 10px 20px;
    font-size: 16px;
    background-color: #2222E0;
    border: none;
    border-radius: 20px;
    color: white;
    cursor: pointer;
}

.action-button:hover {
    background-color: #777;
}
