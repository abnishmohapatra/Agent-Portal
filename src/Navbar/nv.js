import React from 'react';
import './Navbar.css';
import logo from "../InsuranceImages/logosfolder/applogo.png"
import userlogo from "../InsuranceImages/logosfolder/userlogo.png"
function Navbar() {
    return (
        <div className="navbar">
            <div className="navbar-left">
               
                <img src={logo} alt='userlogo' className='logo' />
                <span className="greeting">Inssure</span>
            </div>
            <div className="navbar-middle"></div>
            <div className="navbar-right">
            <img src={userlogo} alt='userloginlogo' className='profile-image' />
                <h1>Name</h1>
                <button className="action-button">Logout</button>
            </div>
        </div>
    );
}
 
export default Navbar;