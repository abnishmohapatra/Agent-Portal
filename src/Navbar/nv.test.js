import React from 'react';
import { render, screen } from '@testing-library/react';
import Navbar from './nv';

describe('Navbar Component', () => {
  beforeEach(() => {
    render(<Navbar />);
  });

  it('renders logo and greeting text', () => {
    const greetingElement = screen.getByText('Inssure');
    expect(greetingElement).toBeInTheDocument();

    const logoElement = screen.getByAltText('userlogo');
    expect(logoElement).toBeInTheDocument();
  });

  it('renders user profile information', () => {
    const nameElement = screen.getByText('Name');
    expect(nameElement).toBeInTheDocument();

    const logoutButton = screen.getByText('Logout');
    expect(logoutButton).toBeInTheDocument();

    const profileImage = screen.getByAltText('userloginlogo');
    expect(profileImage).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
