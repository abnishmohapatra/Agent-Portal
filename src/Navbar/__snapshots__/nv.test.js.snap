// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <div
      class="navbar-left"
    >
      <img
        alt="userlogo"
        class="logo"
        src="applogo.png"
      />
      <span
        class="greeting"
      >
        Inssure
      </span>
    </div>
    <div
      class="navbar-middle"
    />
    <div
      class="navbar-right"
    >
      <img
        alt="userloginlogo"
        class="profile-image"
        src="userlogo.png"
      />
      <h1>
        Name
      </h1>
      <button
        class="action-button"
      >
        Logout
      </button>
    </div>
  </div>
</DocumentFragment>
`;
